draft-ietf-quic-spin-exp.md

```diff
@@ -244,11 +244,8 @@ on the source and destination addresses of the path,
 so that the spin bit is consistently enabled or
 disabled for repeated use of the same path.
 
```

Comment:
Sorry to re-open this conversation but the spin should really be set to zero and not random per packet or anything like that because this makes it harder to impossible to detect valid spinning signals in the network and may render the valid spin signal useless for aggreates.

Reply:
I thought the agreement was that it SHOULD be set to a fixed value, but that value should vary by connection?  Or at least that was my understanding.  I don't think a MUST is enforceable, but a SHOULD seems appropriate.

Reply:
Correction for my side: it doesn't need to be zero but set to a fixed value. However the text saying this got removed. 

This is not enforcable but as a good citizian you should not try to disturb the spin signal provided by other flow and therefore I would even go for a MUST.

Reply:
I thought removing that text was intentional, per the discussion above. If we wanted to change the outcome of that discussion, then I could maybe see a SHOULD, but based on that conversation I can't see it being a MUST.

Reply:
Sorry, I was on holidays and couldn't comment on the discussion above, that why I reopen this conversation. Also Brian is on parental leave, so he was also not able to comment. What I'm saying is that text should not have been removed. We can have more discussion about SHOULD and MUST.

Reply:
I assume that this comment applies to the next paragraph, "When the spin bit is disabled..."

Reply:
@huitema I think so, yeah. 
@mirjak Of course, I'm not saying that we shouldn't keep discussing, I just didn't want to copy/paste everything from above, so I referred back to that. 😃

Reply:
I don't have a strong opinion, but at least Marcus seemed to believe that a random signal can be filtered out just as easily as a fixed zero or one.

Reply:
Yes, it's not hard to filter out, if your the goal/use case is to only get valid samples. However if you are interested in e.g. detecting large amount of re-ordering which will cause invalid samples within a flow that otherwise provides a valid spin signal, that would be harder to detect because re-ordering will also look like random noise. Therefore I would like to see us recommening a constant value. Of course we cannot enforce this and still have to handle noise, however, if the noise is low enough, it should still be fine.

Reply:
I see, your point is that it will be harder to distinguish between those who are spinning but encountering reordering and those who are just setting the value randomly.  Doesn't that wash out with time?  I mean, you need time to recover the spin signal anyway, so how much time do you need to discover that you are getting rubbish?

Reply:
I guess that depends on the heuristic you use. However, for a single flow you don't really have a recovery phase. Either you have a valid sample or not. If the base RTT didn't change, you should get a valid sample right after the re-ordering event is over. The interesting part is to observe the number of invalid samples. So for a single, long-lasting flow that you know has provided a valid spin signal previously, there is also no problem. What I'm worried about is if you monitor the invalid sample rate for an aggregate; especially if you have a lot of short flows, as you need more time to detect that a flow is spinning randomly than when using a constant value. Thus for short flows you may not be able make a decision in time and you have to filter them out. Given we have a lot of short flows on the internet, that can potentially strongly reduce your sample rate. Actually I would even prefer that disabled flows set the bit to 0 while enabled clients start with 1 because that would make it possible to have valid spin signals from he beginning. Still for any constant value, you can at least make a decision within about one RTT.

I mean this is certainly something to experiment with and the actual impact really depends on how many flows do what as well as the traffic patter of the flows. What I’m saying is that a constant value makes measurability at least a bit easier, and if there are no further concern about using a constant value, I would prefer if we recommend that.

My understanding of ekr’s initial comment was that he wanted to leave it open how to set the bit when disabled, as he didn’t see an advantage in (over-)specifying that behavior. Others were actually concerned that setting it randomly (if the randomness is not sufficiently provided) could enable fingerprinting. As such I don’t see a reason to not recommend a constant value when disabled.

Reply:
I don't agree with this. I see a *disadvantage* in specifying the behavior to be fixed, which is that it makes discrimination against endpoints which used fixed values easier.